src/main/java/com/getindata/connectors/http/internal/sink/httpclient/status/ComposeHttpStatusCodeChecker.java

```diff
@@ -0,0 +1,128 @@
+
+    private static final Set<TypeStatusCodeChecker> DEFAULT_ERROR_CODES =
+        Set.of(
+            new TypeStatusCodeChecker(HttpType.CLIENT_ERROR),
+            new TypeStatusCodeChecker(HttpType.SERVER_ERROR)
+        );
+
+    private static final int MIN_HTTP_ERROR_CODE = 100;
+
+    /**
+     * Set of {@link HttpStatusCodeChecker} for white listed status codes.
+     */
+    private final Set<WhiteListHttpStatusCodeChecker> excludedCodes;
+
+    /**
+     * Set of {@link HttpStatusCodeChecker} that check status code againts value match or {@link
+     * HttpType} match.
+     */
+    private final Set<HttpStatusCodeChecker> errorCodes = new HashSet<>();
+
+    public ComposeHttpStatusCodeChecker(Properties properties) {
+        excludedCodes = prepareWhiteList(properties);
+        prepareErrorCodes(properties);
+    }
+
+    /**
+     * Checks whether given status code is considered as a error code.
+     * This implementation checks if status code matches any single value mask like "404"
+     * or http type mask such as "4XX". Code that matches one of those masks and is not on a
+     * white list will be considered as error code.
+     * @param statusCode http status code to assess.
+     * @return true if status code is considered as error or false if not.
+     */
+    public boolean isErrorCode(int statusCode) {
+
+        Preconditions.checkArgument(
+            statusCode >= MIN_HTTP_ERROR_CODE,
+            String.format(
+                "Provided invalid Http status code %s,"
+                    + " status code should be equal or bigger than 100.",
+                statusCode)
+        );
+
+        boolean isWhiteListed = excludedCodes.stream()
+            .anyMatch(check -> check.isWhiteListed(statusCode));
+
+        if (isWhiteListed) {
+            return false;
+        } else {
+            return errorCodes.stream()
+                .anyMatch(httpStatusCodeChecker -> httpStatusCodeChecker.isErrorCode(statusCode));
+        }
+    }
+
+    private void prepareErrorCodes(Properties properties) {
+        String sErrorCodes =
+            properties.getProperty(HttpConnectorConfigConstants.HTTP_ERROR_CODES_LIST, "");
+
+        if (StringUtils.isNullOrWhitespaceOnly(sErrorCodes)) {
+            errorCodes.addAll(DEFAULT_ERROR_CODES);
+        } else {
+            String[] splitCodes = sErrorCodes.split(HttpConnectorConfigConstants.ERROR_CODE_DELIM);
+
+            if (splitCodes.length == 0) {
+                errorCodes.addAll(DEFAULT_ERROR_CODES);
+            } else {
+                prepareErrorCodes(splitCodes);
+            }
+        }
+    }
+
+    /**
+     * Process given array of status codes and assign them to
+     * {@link SingleValueHttpStatusCodeChecker} for full codes such as 100, 404 etc. or to
+     * {@link TypeStatusCodeChecker} for codes that were constructed with "XX" mask
+     */
+    private void prepareErrorCodes(String[] statusCodes) {
```

Comment:
Yeah, 
I don't have better idea for the name. Thing is that this method (and also the parent one) job is to "prepare error codes" collections.

Also I have spotted this kind of patter (hope not anti pattern) in other projects. 
I got your point but on the other hand unless we would come up with a really good two different names for those methods, this might be just a case of method overloading as well :)

Reply:
Ok, I won't argue about that, let's keep it that way :)
It's just that I have some irrational anxiety about two overloaded methods that aren't returning anything and rely solely on side effects 😛 On the other hand, both are just adding values to the `HashSet` based on the given input, so it should not be harmful in any way :d Considering all of this, both can have the same name 😄

Reply:
the " two overloaded methods that aren't returning anything and rely solely on side effects" I buy.
I have changed the code so now both methods are returning a Set that is assigned to final errorCodes class field.